torch/csrc/autograd/engine.cpp

```diff
@@ -545,7 +559,13 @@ auto Engine::evaluate_function(NodeTask& task) -> void {
   }
 
   int num_outputs = outputs.size();
-  if (num_outputs == 0) return; // Don't even acquire the mutex
+  if (num_outputs == 0) {
+    // Records leaf stream (if applicable)
```

Comment:
I'm having a little trouble understanding, what is a "leaf stream"? (I know you've already been through a bit of review, but this my first time reading the patch haha)

Reply:
See larger comment in conversation. Leaf streams are the streams of the leaf nodes in the graph, so syncing with them is sufficient to ensure that backward has completed all its work.

Reply:
Would be nice to have this nice comment somewhere in the code as a Note, and then cross-referenced from the relevant locations :)

Reply:
Will do.

Reply:
Added "Streaming backwards" note.